client/blocks/login/social.jsx

```diff
@@ -2,134 +2,103 @@ import { Card } from '@automattic/components';
+		trackLoginAndRememberRedirect: PropTypes.func.isRequired,
 		socialServiceResponse: PropTypes.object,
 		shouldRenderToS: PropTypes.bool,
 		magicLoginLink: PropTypes.string,
 		qrLoginLink: PropTypes.string,
 		isSocialFirst: PropTypes.bool,
+		lastUsedAuthenticationMethod: PropTypes.string,
+		resetLastUsedAuthenticationMethod: PropTypes.func,
 	};
 
 	static defaultProps = {
 		shouldRenderToS: false,
 	};
 
-	handleLogin = ( result ) => {
-		const socialLoginUsed = window.sessionStorage.getItem( 'social_login_used' );
-
-		if ( ! result || socialLoginUsed !== result.service ) {
-			return;
-		}
-
-		const { onSuccess, loginUser } = this.props;
-
-		let redirectTo = this.props.redirectTo;
-
-		// load persisted redirect_to url from session storage, needed for redirect_to to work with google redirect flow
-		if ( ! redirectTo ) {
-			redirectTo = window.sessionStorage.getItem( 'login_redirect_to' );
-		}
-
-		window.sessionStorage.removeItem( 'login_redirect_to' );
-		window.sessionStorage.removeItem( 'social_login_used' );
-
-		loginUser( result, redirectTo ).then(
-			() => {
-				this.recordEvent( 'calypso_login_social_login_success', result.service );
-				onSuccess();
-			},
-			( error ) => {
-				if ( error.code === 'user_exists' || error.code === 'unknown_user' ) {
-					this.props.createSocialUserFailed( result, error, 'login' );
-					return;
-				}
-
-				this.recordEvent( 'calypso_login_social_login_failure', result.service, {
-					error_code: error.code,
-					error_message: error.message,
-				} );
-			}
-		);
-	};
-
-	trackLoginAndRememberRedirect = ( event ) => {
-		const service = event.currentTarget.getAttribute( 'data-social-service' );
-		this.recordEvent( 'calypso_login_social_button_click', service );
-
-		window.sessionStorage.setItem( 'social_login_used', service );
-
-		if ( this.props.redirectTo && typeof window !== 'undefined' ) {
-			window.sessionStorage.setItem( 'login_redirect_to', this.props.redirectTo );
-		}
-	};
-
-	recordEvent = ( eventName, service, params ) =>
-		this.props.recordTracksEvent( eventName, {
-			social_account_type: service,
-			...params,
-		} );
+	socialLoginButtons = [
+		{
+			service: 'google',
+			button: (
+				<GoogleSocialButton
+					responseHandler={ this.props.handleLogin }
+					onClick={ this.props.trackLoginAndRememberRedirect }
+					key={ 1 }
+					isLogin
+				/>
+			),
+		},
+		{
+			service: 'apple',
+			button: (
+				<AppleLoginButton
+					responseHandler={ this.props.handleLogin }
+					onClick={ this.props.trackLoginAndRememberRedirect }
+					socialServiceResponse={ this.props.socialServiceResponse }
+					key={ 2 }
+					isLogin
+				/>
+			),
+		},
+		{
+			service: 'github',
+			button: (
+				<GithubSocialButton
+					responseHandler={ this.props.handleLogin }
+					onClick={ this.props.trackLoginAndRememberRedirect }
+					socialServiceResponse={ this.props.socialServiceResponse }
+					key={ 3 }
+					isLogin
+				/>
+			),
+		},
+		{
+			service: 'magic',
```

Comment:
Let's rename this `magic-login`. That should fix the "Continue with Email" button issue where it doesn't appear if you previously logged in with the magic login link option.